tools/mac-distribution/main.go

```diff
@@ -0,0 +1,168 @@
+}
+
+type AppBundleCmd struct {
+	AppBinary string `arg:"" help:"Binary to use as the main executable for the app bundle."`
+	Skeleton  string `arg:"" help:"Skeleton directory to use as the base for the app bundle."`
+
+	Entitlements string `flag:"" help:"Entitlements file to use for the app bundle."`
+	BundleID     string `flag:"" required:"" help:"Bundle ID is a unique identifier for the app bundle. Required for notarization."`
+
+	NotaryCmd
+}
+
+type PackageInstallerCmd struct {
+	RootPath          string `arg:"" help:"Path to the root directory of the package installer."`
+	PackageOutputPath string `arg:"" help:"Path to the output package installer."`
+
+	InstallLocation string `flag:"" required:"" help:"Location where the package contents will be installed."`
+	BundleID        string `flag:"" required:"" help:"Bundle ID is a unique identifier for the package installer."`
+	ScriptsDir      string `flag:"" help:"Path to the scripts directory. Contains preinstall and postinstall scripts."`
+	Version         string `flag:"" help:"Version of the package. Used in determining upgrade behavior."`
+
+	NotaryCmd
+}
+
+func main() {
+	var cli CLI
+
+	kctx := kong.Parse(&cli)
+	err := kctx.Run()
+	kctx.FatalIfErrorf(err)
+}
+
+func (c *AppBundleCmd) Run(cli *CLI) error {
+	pkg, err := appbundle.NewPackager(
+		appbundle.Info{
+			Skeleton:     c.Skeleton,
+			Entitlements: c.Entitlements,
+			AppBinary:    c.AppBinary,
+		},
+		appbundle.WithLogger(log),
+		appbundle.WithNotaryTool(c.notaryTool),
+		appbundle.WithBundleID(c.BundleID),
+	)
+	if err != nil {
+		return err
+	}
+
+	return pkg.Package()
+}
+
+func (c *PackageInstallerCmd) Run(cli *CLI) error {
+	pkg, err := packageinstaller.NewPackager(
+		packageinstaller.Info{
+			RootPath:        c.RootPath,
+			InstallLocation: c.InstallLocation,
+			OutputPath:      c.PackageOutputPath,
+			ScriptsDir:      c.ScriptsDir,
+			BundleID:        c.BundleID,
+		},
+		packageinstaller.WithLogger(log),
+		packageinstaller.WithNotaryTool(c.notaryTool),
+	)
+	if err != nil {
+		return err
+	}
+
+	return pkg.Package()
+}
+
+func (c *NotarizeCmd) Run(cli *CLI) error {
+	return c.notaryTool.NotarizeBinaries(c.Files)
+}
+
+func (g *NotaryCmd) AfterApply() error {
+	// Dry run if no credentials are provided
+	credsMissing := g.AppleUsername == "" || g.ApplePassword == "" || g.SigningID == "" || g.TeamID == ""
+
+	if !g.DryRun && credsMissing {
+		return errors.New("notarization credentials required, use --dry-run to skip")
+	}
+
+	if g.CI && g.DryRun {
+		return errors.New("--dry-run mode cannot be used in CI")
+	}
+
+	extraOpts := []notarize.Opt{}
+
+	if g.DryRun {
+		extraOpts = append(extraOpts, notarize.DryRun())
+	}
+
+	t, err := notarize.NewTool(
+		notarize.Creds{
+			AppleUsername:   g.AppleUsername,
+			ApplePassword:   g.ApplePassword,
+			SigningIdentity: g.SigningID,
+			TeamID:          g.TeamID,
+		},
+		append(
+			extraOpts,
+			notarize.MaxRetries(g.Retry),
+			notarize.WithLogger(log),
```

Comment:
Can these be set when initializing `extraOpts`?

    extraOpts := []notarize.Opt{notarize.MaxRetries(g.Retry), notarize.WithLogger(log)}

That would simplify this `NewTool()` expression somewhat,